src/test/ui/parser/paren-after-qualified-path-in-match.stderr

```diff
@@ -2,7 +2,9 @@ error: unexpected `(` after qualified path
   --> $DIR/paren-after-qualified-path-in-match.rs:3:27
    |
 LL |         <T as Trait>::Type(2) => (),
-   |                           ^ unexpected `(` after qualified path
+   |         ------------------^ unexpected `(` after qualified path
+   |         |
+   |         the qualified path
```

Comment:
I could make this recover, but then the `vec![...] =>` recovery has to go away (and I think it should, because it is causing problems in https://github.com/rust-lang/rust/pull/66364 as well).

Reply:
It'd be a shame to lose it entirely, but if it improves this case it might be worth it. For the vec case if we lose the suggestion at least it should recover gracefully.

Reply:
Is there a way to have the `vec![...]` work with resolve instead of first checking if there is an error in expansion (i.e. check if the path is actually `{core,std}::vec`)? Perhaps we can use `diagnostic_item`s? It feels like the current encoding of the error is problematic in terms of infrastructure. What do you think @petrochenkov?

Reply:
The `vec![...]` hack is really bad, but I cannot answer whether it can be done it a better way without some investigation.